sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/AzureLogAnalyticsDataFeedSource.java

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.ai.metricsadvisor.models;
+
+import com.azure.core.annotation.Immutable;
+
+/** The AzureLogAnalyticsDataFeedSource model. */
+@Immutable
+public final class AzureLogAnalyticsDataFeedSource extends DataFeedSource {
+    /*
+     * The tenant id of service principal that have access to this Log
+     * Analytics
+     */
+    private final String tenantId;
+
+    /*
+     * The client id of service principal that have access to this Log
+     * Analytics
+     */
+    private final String clientId;
+
+    /*
+     * The client secret of service principal that have access to this Log
+     * Analytics
+     */
+    private final String clientSecret;
+
+    /*
+     * The workspace id of this Log Analytics
+     */
+    private final String workspaceId;
+
+    /*
+     * The KQL (Kusto Query Language) query to fetch data from this Log
+     * Analytics
+     */
+    private final String query;
+
+    /**
+     * Create a AzureLogAnalyticsDataFeedSource instance.
+     *
+     * @param tenantId The tenant id of service principal that have access to this Log Analytics.
+     * @param clientId The client id of service principal that have access to this Log Analytics.
+     * @param clientSecret The client secret of service principal that have access to this Log Analytics.
+     * @param workspaceId the query script.
+     * @param query the KQL (Kusto Query Language) query to fetch data from this Log
+     * Analytics.
+     */
+    public AzureLogAnalyticsDataFeedSource(String tenantId, String clientId, String clientSecret,
+                                           String workspaceId, String query) {
+        this.tenantId = tenantId;
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
+        this.workspaceId = workspaceId;
+        this.query = query;
+    }
+
+    /**
+     * Get the tenantId property: The tenant id of service principal that have access to this Log Analytics.
+     *
+     * @return the tenantId value.
+     */
+    public String getTenantId() {
+        return this.tenantId;
+    }
+
+    /**
+     * Get the clientId property: The client id of service principal that have access to this Log Analytics.
+     *
+     * @return the clientId value.
+     */
+    public String getClientId() {
+        return this.clientId;
+    }
+
+    /**
+     * Get the clientSecret property: The client secret of service principal that have access to this Log Analytics.
+     *
+     * @return the clientSecret value.
+     */
+    public String getClientSecret() {
+        return this.clientSecret;
+    }
```

Comment:
We might have to hide few properties from public API (such as secrets), we don't have the full list as we're waiting for the service team to confirm. We can do that once we get a response, not a blocker for this pr.